src/node/SetupNodeServerApi.ts

```diff
@@ -0,0 +1,47 @@
+import { defaultMaxListeners, setMaxListeners } from 'node:events'
+import { SetupServerApi } from './SetupServerApi'
+import { SetupServer } from './glossary'
+import { isNodeExceptionLike } from './utils/isNodeExceptionLike'
+
+export class SetupNodeServerApi extends SetupServerApi implements SetupServer {
```

Comment:
I like this approach. A couple of thoughts to make it stellar:

1. Don’t introduce a new class.
2. Add a protected `onRequest` method to the existing `SetupServerApi` class. Do nothing in that method.
3. In `src/node/index.ts`, create a new subclass from `SetupServerApi`. In that subclass, invoke the abort listener logic directly (no need to abstract to a method). This way, the import to `node:events` will be scoped to `node/index.ts` and won’t leak to `/native` on `SetupServerApi.ts` imports there.
4. No changes are necessary for the React Native setup API. 

So, this is basically identical to what you did just inlining the new derived class directly in` src/node/index.ts`. What do you think about this?

Reply:
My main concern here is that `SetupApi` classes are meant to cover functionality. Here, we sort of address the extension issue that's import-specific (if we import `node:events` in the base `SetupApi`, any consumer automatically gets that import, which is not what we want). An internal issue.

Reply:
yep, makes sense.  

https://github.com/mswjs/msw/pull/1885/commits/1e44a5b3bcc2b12f6129bac3c49db8dbc4f5a458 might be what you're suggesting (just keeping from a separate file?)